syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

package communication;

message InvoicingRequest {
  ProtoUser user = 1;
  uint64 price = 2;
}

message InvoicingResponse {
  StatusCode status = 1;
}

message ProtoUser {
  string id = 1;
  string firstName = 2;
  string lastName = 3;
  string email = 4;
  Address address = 5;
}

message Address {
  string street = 1;
  string houseNumber = 2;
  string postalCode = 3;
  string city = 4;
  string country = 5;
}

enum UnitPriceType {
  FARE_ZONE = 0;
  KM = 1;
  MIN = 2;
  DAY = 3;
  WEEK = 4;
  MONTH = 5;
  YEAR = 6;
}

enum TransporterType {
  SHARED_AUTO = 0;
  REGIONAL_BAHN = 1;
  AUTO = 2;
  S_BAHN = 3;
  TAXI = 4;
  BUS = 5;
  E_ROLLER = 6;
  SCOOTER = 7;
  BICYCLE = 8;
  TRAM = 9;
}

enum StatusCode {
  FAILED = 0;
  SUCCESSFUL = 1;
}

enum BookingType{
  RENTAL = 0;
  RIDE = 1;
}

enum InvoicingClerkType {
  UNKNOWN = 0;
  DEBIT = 1;
  PAYPAL = 2;
  VISA = 3;
  MASTER = 4;
  GIROPAY = 5;
  KLARNA = 6;
}
